src/parser/inlines.rs

```diff
@@ -1548,6 +1571,118 @@ impl<'a, 'r, 'o, 'd, 'i, 'c, 'subj> Subject<'a, 'r, 'o, 'd, 'i, 'c, 'subj> {
         }
     }
 
+    // Handles wikilink syntax
+    //   [[link text|url]]
+    //   [[url|link text]]
+    pub fn handle_wikilink(&mut self) -> Option<&'a AstNode<'a>> {
+        let startpos = self.pos;
+        let (url, title) = self.wikilink_url_title();
+
+        url?;
```

Comment:
```suggestion
        let url = url.unwrap();
```

(and then change below `url.unwrap()`s to just `url`)

Reply:
But isn't `url?;` the same as

```
if url.is_none
    return None;
end
```

which would be needed since, at least in this incarnation, url can be `None`. Unwrapping it would panic.

Reply:
Ah, right; in that case I'd suggest `let url = url?;` here, to avoid the need to unwrap later while still having the monadic return kind of behaviour.

Reply:
Makes sense, changed